protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -73,6 +73,35 @@ definitions:
       emitted_at:
         description: "when the data was emitted from the source. epoch in millisecond."
         type: integer
+      meta:
+        description: Information about this record added mid-sync
+        "$ref": "#/definitions/AirbyteRecordMessageMeta"
+  AirbyteRecordMessageMeta:
+    type: object
+    additionalProperties: true
+    properties:
+      errors:
+        description: Key-Value pairs of problems for properties of this record
+        type: array
+        items:
+          "$ref": "#/definitions/AirbyteRecordMessageMetaErrors"
+  AirbyteRecordMessageMetaErrors:
+    type: object
+    additionalProperties: true
+    required:
+      - key
+      - message
+    properties:
+      key:
+        type: string
```

Comment:
Of your two additional proposal I think I like the json path version and that it helps the user identify exactly what the problem is, however I think in the 1s1t world view where object is still a just a column I would go with the first one (the one presented in the doc and this pr) since I'm still slightly concerned about compute resources and recursing through the JSON to find the offending path(s)

Reply:
hm. what _is_ our plan for file-based destinations? today in destination-s3 we fully convert the jsonschema to an avro schema... do we want to eventually swap that to just put a json string field for nonprimitive types? In theory, 1s1t only directly applies to sqlish destinations

either way though - I don't think we can avoid the recursion. Redshift will error if any string in the entire blob is >64KiB, and we can't really detect that without fully traversing the tree

Reply:
I think the long-term plan for file destinations is that they end up looking a lot like Dv2 destinations - top level properties get typed, and we add a few extra columns, one of which is `airbyte_meta`

Reply:
It's sort of interesting that the protocol doesn't _really_ care which solution we pick for sub-keys, assuming it can be string-represented.  That said, I also like the jsonPath version.  I'll add an example to the description of this PR.

Reply:
let's add a comment/description to the protocol yaml :D